packages/core/src/shared/logger/toolkitLogger.ts

```diff
@@ -113,22 +120,19 @@ export class ToolkitLogger extends BaseLogger implements vscode.Disposable {
     }
 
     /* Format the message with topic header */
-    private addTopicToMessage(message: string | Error): string | object {
-        /*We shouldn't print unknow before current logging calls are migrated
-         * TODO: remove this once migration of current calls is completed
-         */
-        if (this.topic === 'unknown') {
-            return message
-        }
-        const topic = `${this.topic}: `
+    private addTopicToMessage(message: string | Error): string | ErrorLog {
+        const topicPrefix = `${this.topic}: `
         if (typeof message === 'string') {
-            return topic + message
-        } else if (message instanceof Error) {
-            const errorWithTopic = {
-                topic,
-                message,
+            /*We shouldn't print unknow before current logging calls are migrated
+             * TODO: remove this once migration of current calls is completed
+             */
+            if (this.topic === 'unknown') {
+                return message
             }
-            return errorWithTopic
+            return topicPrefix + message
```

Comment:
See the second part of my comment here: https://github.com/aws/aws-toolkit-vscode/pull/5610#discussion_r1765293320

We can get rid of the split down below